### ISSUE TITLE
Outstanding Issue from PR #222

### ISSUE THREAD

**Issue by enoren5** [enoren5/tarot_juicer#223]
When I first pulled  changes from PR #222 to test, I kept encountering a 404 saying that the page doesn't exist. It took me thirty minutes to realize that the default Bool value for the new `is_published` attributes were set to `False`. So I located every `is_published` class attribute and switched the BooleanFields defaults to `True`. That's how it should be. I also noticed that you appended the `is_published` attributes to the very bottom of the model class definitions. As a result, the switch on the Admin Dashboard is all the way at the bottom which makes it hard to locate and easy to overlook. So I moved the `is_published` attributes closer to the top. 

After making the migrations and then migrating the local `db.sqlite3`, everything seems to work now. Thank you.

Although I noticed after merging the changes into `master` and pushing the changes to my staging/testing pipeline on Heroku, my app is throwing "ProgrammingErrors" and "django.db.utils.OperationalErrors". 

See here:

![Screenshot from 2022-11-08 01-53-21](https://user-images.githubusercontent.com/6686723/200495261-fdf5fa8a-9426-4cf7-aa80-b589df1e45e8.png)

Any idea what may be causing this? I Googled around and the many recurring themes in upvoted Stack Overflow answers suggest that the database needs to be migrated. But the automated Heroku Django buildback does this when deployed. It's part of the built-in routine. I tried logging in using Heroku CLI in my shell and ran makemigrations for the primary `generators` app. Below you can see  was the output of me trying to manually migrate the remote Heroku db. Notice that it often refers to db.qlite3. That is odd because remotely Heroku should be using PostreSQL. Any idea what might be happening here?

```
(heroku remote environment) ~ $ python manage.py makemigrations generators

Secure Mode Disabled: DEBUG MODE IS TRUE


Database Config: {'default': {'NAME': '/app/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}

Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: accounts_authtoggle

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 21, in <module>
    main()
  File "/app/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 393, in execute
    self.check()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/tarot_juicer/urls.py", line 12, in <module>
    path('', include('accounts.urls')),
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/accounts/urls.py", line 2, in <module>
    from . import views  # , include
  File "/app/accounts/views.py", line 14, in <module>
    SESSION_TIMEOUT = AuthToggle.objects.first()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 674, in first
    for obj in (self if self.ordered else self.order_by('pk'))[:1]:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 280, in __iter__
    self._fetch_all()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: accounts_authtoggle
```

**Comment by enoren5** [comment]
Abu elsewhere suggested that I run `$ python manage.py migrate`. That sounded very silly and a novice thing to suggest  because I am wondering: "why would migrating the database work when making the migrations failed?" Making migrations needs to run first. So if that is failing, then why would migrating work? That was my thought. But I ran `$ python manage.py migrate` as Abu sugggested and it didn't work either (just as I thought). Here is the broken output:

```
~ $ python manage.py migrate

Secure Mode Disabled: DEBUG MODE IS TRUE


Database Config: {'default': {'NAME': '/app/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}

Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: accounts_authtoggle

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 21, in <module>
    main()
  File "/app/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 75, in handle
    self.check(databases=[database])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/tarot_juicer/urls.py", line 12, in <module>
    path('', include('accounts.urls')),
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/accounts/urls.py", line 2, in <module>
    from . import views  # , include
  File "/app/accounts/views.py", line 14, in <module>
    SESSION_TIMEOUT = AuthToggle.objects.first()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 674, in first
    for obj in (self if self.ordered else self.order_by('pk'))[:1]:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 280, in __iter__
    self._fetch_all()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: accounts_authtoggle
~ $ 
```

**Comment by abubakarA-Dot** [comment]
> When I first pulled changes from PR #222 to test, I kept encountering a 404 saying that the page doesn't exist. It took me thirty minutes to realize that the default Bool value for the new `is_published` attributes were set to `False`. So I located every `is_published` class attribute and switched the BooleanFields defaults to `True`. That's how it should be. I also noticed that you appended the `is_published` attributes to the very bottom of the model class definitions. As a result, the switch on the Admin Dashboard is all the way at the bottom which makes it hard to locate and easy to overlook. So I moved the `is_published` attributes closer to the top.
> 
> After making the migrations and then migrating the local `db.sqlite3`, everything seems to work now. Thank you.
> 
> Although I noticed after merging the changes into `master` and pushing the changes to my staging/testing pipeline on Heroku, my app is throwing "ProgrammingErrors" and "django.db.utils.OperationalErrors".
> 
> See here:
> 
> ![Screenshot from 2022-11-08 01-53-21](https://user-images.githubusercontent.com/6686723/200495261-fdf5fa8a-9426-4cf7-aa80-b589df1e45e8.png)
> 
> Any idea what may be causing this? I Googled around and the many recurring themes in upvoted Stack Overflow answers suggest that the database needs to be migrated. But the automated Heroku Django buildback does this when deployed. It's part of the built-in routine. I tried logging in using Heroku CLI in my shell and ran makemigrations for the primary `generators` app. Below you can see was the output of me trying to manually migrate the remote Heroku db. Notice that it often refers to db.qlite3. That is odd because remotely Heroku should be using PostreSQL. Any idea what might be happening here?
> 
> ```
> (heroku remote environment) ~ $ python manage.py makemigrations generators
> 
> Secure Mode Disabled: DEBUG MODE IS TRUE
> 
> 
> Database Config: {'default': {'NAME': '/app/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}
> 
> Traceback (most recent call last):
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
>     return Database.Cursor.execute(self, query, params)
> sqlite3.OperationalError: no such table: accounts_authtoggle
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/app/manage.py", line 21, in <module>
>     main()
>   File "/app/manage.py", line 17, in main
>     execute_from_command_line(sys.argv)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
>     utility.execute()
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
>     self.fetch_command(subcommand).run_from_argv(self.argv)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
>     self.execute(*args, **cmd_options)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 393, in execute
>     self.check()
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
>     all_issues = checks.run_checks(
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
>     new_errors = check(app_configs=app_configs, databases=databases)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
>     all_namespaces = _load_all_namespaces(resolver)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
>     url_patterns = getattr(resolver, 'url_patterns', [])
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
>     res = instance.__dict__[self.name] = self.func(instance)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
>     patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
>     res = instance.__dict__[self.name] = self.func(instance)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
>     return import_module(self.urlconf_name)
>   File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/app/tarot_juicer/urls.py", line 12, in <module>
>     path('', include('accounts.urls')),
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/conf.py", line 34, in include
>     urlconf_module = import_module(urlconf_module)
>   File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/app/accounts/urls.py", line 2, in <module>
>     from . import views  # , include
>   File "/app/accounts/views.py", line 14, in <module>
>     SESSION_TIMEOUT = AuthToggle.objects.first()
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
>     return getattr(self.get_queryset(), name)(*args, **kwargs)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 674, in first
>     for obj in (self if self.ordered else self.order_by('pk'))[:1]:
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 280, in __iter__
>     self._fetch_all()
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 1324, in _fetch_all
>     self._result_cache = list(self._iterable_class(self))
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
>     results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
>     cursor.execute(sql, params)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
>     return super().execute(sql, params)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
>     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
>     return executor(sql, params, many, context)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
>     with self.db.wrap_database_errors:
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
>     raise dj_exc_value.with_traceback(traceback) from exc_value
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
>   File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
>     return Database.Cursor.execute(self, query, params)
> django.db.utils.OperationalError: no such table: accounts_authtoggle
> ```

I have noticed that you are trying to migrate the generators app but you are facing migrations issues related to accounts, landings but you are trying to apply migrations for the generators app only. I would suggest you, remove the generators app after `python manage.py makemigrations` command  or you have to migrate the apps one by one starting from accounts and landings.

**Comment by enoren5** [comment]
Hi @abubakarA-Dot: Here is the output of `$ python manage.py makemigrations` (without specifying an app) and `$ python manage.py migrate`:

```
~ $ python manage.py makemigrations

Secure Mode Disabled: DEBUG MODE IS TRUE


Database Config: {'default': {'NAME': '/app/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}

Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: accounts_authtoggle

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 21, in <module>
    main()
  File "/app/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 393, in execute
    self.check()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/tarot_juicer/urls.py", line 12, in <module>
    path('', include('accounts.urls')),
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/accounts/urls.py", line 2, in <module>
    from . import views  # , include
  File "/app/accounts/views.py", line 14, in <module>
    SESSION_TIMEOUT = AuthToggle.objects.first()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 674, in first
    for obj in (self if self.ordered else self.order_by('pk'))[:1]:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 280, in __iter__
    self._fetch_all()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: accounts_authtoggle
~ $ python manage.py migrate

Secure Mode Disabled: DEBUG MODE IS TRUE


Database Config: {'default': {'NAME': '/app/db.sqlite3', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'CONN_MAX_AGE': 600, 'ENGINE': 'django.db.backends.sqlite3'}}

Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: accounts_authtoggle

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 21, in <module>
    main()
  File "/app/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 75, in handle
    self.check(databases=[database])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
    all_issues = checks.run_checks(
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/core/checks/urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/tarot_juicer/urls.py", line 12, in <module>
    path('', include('accounts.urls')),
  File "/app/.heroku/python/lib/python3.10/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/app/.heroku/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/app/accounts/urls.py", line 2, in <module>
    from . import views  # , include
  File "/app/accounts/views.py", line 14, in <module>
    SESSION_TIMEOUT = AuthToggle.objects.first()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 674, in first
    for obj in (self if self.ordered else self.order_by('pk'))[:1]:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 280, in __iter__
    self._fetch_all()
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/app/.heroku/python/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: accounts_authtoggle
~ $ 
```

**Comment by enoren5** [comment]
I turned on the `is_published` Bool switches for each of the 4 cards on the testing db. Then there was an error indicating "4 instances when there should only be one" (or something to this effect). I was going to write up a new issue and brand new Stack Overflow question but then I reviewed @abubakarA-Dot's comments in his PR #222 where he clarifies: "If `is_published` feature is on for any single generator card, it turns the whole tarot cards on." So I switched three of the four `is_published` Bools to off and the `generators` app began working perfectly. No more errors. 

I am glad our project is running now. 

But with the way `is_published` works currently for the generators app could lead to confusion down the road. So I tried commenting out the `is_published` attribute declaration in the generators class model but Django didn't like that:
![image](https://user-images.githubusercontent.com/6686723/208284299-5f53024b-5376-4f07-9a97-66acae6da0e3.png)

Before we can move on to the next and final requirement for this project, this is the final task: Abu could you remove `is_published` attribute from the generators model in such a way that Django won't throw this FieldError Exception? I think we can do without `is_published` in the `Generator` model.

**Comment by abubakarA-Dot** [comment]
> I turned on the `is_published` Bool switches for each of the 4 cards on the testing db. Then there was an error indicating "4 instances when there should only be one" (or something to this effect). I was going to write up a new issue and brand new Stack Overflow question but then I reviewed @abubakarA-Dot's comments in his PR #222 where he clarifies: "If `is_published` feature is on for any single generator card, it turns the whole tarot cards on." So I switched three of the four `is_published` Bools to off and the `generators` app began working perfectly. No more errors.
> 
> I am glad our project is running now.
> 
> But with the way `is_published` works currently for the generators app could lead to confusion down the road. So I tried commenting out the `is_published` attribute declaration in the generators class model but Django didn't like that: ![image](https://user-images.githubusercontent.com/6686723/208284299-5f53024b-5376-4f07-9a97-66acae6da0e3.png)
> 
> Before we can move on to the next and final requirement for this project, this is the final task: Abu could you remove `is_published` attribute from the generators model in such a way that Django won't throw this FieldError Exception? I think we can do without `is_published` in the `Generator` model.

This field error occurs when there is no attribute in the model but you still try to call that attribute in your logic( in views.py or elsewhere). so you also need to comment out the is_published logic from generators app views.py ( from `def tarot_key(request, generator_number):` ) as well. after removing it. It will fix the issue.

**Comment by enoren5** [comment]
Resolved in #225. Thanks, @abubakarA-Dot!
